lib/diagnostic.js

```diff
@@ -19,35 +19,47 @@
 //
 //     `true` or `false`
 //
-let access = null; // replace null with your response
+let access = true; // replace null with your response
 //
 
 // Which scope(s) may access the variable `laserQuantity`? Choose one or many
 // of the following:
 //
 //     `['alderaan', 'deathStar', 'fire', 'global']`
 //
-let scopes = [/* your response here */];
+let scopes = ['deathStar', 'fire'];
 //
 
 // Study the following code. Then respond to the questions that follow.
 //
 // NOTE: Line numbers are annotated line numbers, not editor line numbers.
-
-/*1 */ alderaan = { name: 'Alderaan' };
-/*2 */ deathStar = {
-/*3 */   fire: function fire(){
-/*4 */     return this.name;
-/*5 */   }
-/*6 */ };
+/*1 */
+/*2 */
+/*3 */
+/*4 */
+/*5 */
+/*6 */
+alderaan = { name: 'Alderaan' };
+deathStar = {
+  fire: function fire(){
+    return this.name;
+  }
+};
 
 // Write the code that would call the `deathStar`'s `fire` method in the
 // context of `alderaan`. In other words, we want to *reference* the `fire`
 // method and temporarily bind it (or, change `this`) to `alderaan` before
 // executing it.
 //
+
+//this runs but doesn't answer the question
+alderaan.fire = deathStar.fire;
+alderaan.fire();
+
+//real answer below
 let fireOnAlderaan = function () {
-  // your code here
+  let Fire = deathStar.fire;
+  Fire.bind(alderaan);
```

Comment:
close but you have to use .call instead of .bind. bind returns a bound function, but doesn't call it. call binds and calls the function